src/librt/primitives/annot/annot.c

```diff
@@ -951,6 +951,10 @@ rt_annot_import5(struct rt_db_internal *ip, const struct bu_external *ep, const
 		bu_vls_init(&tsg->label);
 		bu_vls_strcpy(&tsg->label, (const char*)ptr);
 		ptr += bu_vls_strlen(&tsg->label) + 1;
+		tsg->txt_size = ntohl(*(uint32_t*)ptr);
+		ptr += SIZEOF_NETWORK_DOUBLE;
```

Comment:
Is this right?  Maybe related to what @drossberg was saying.  You called ntohl to unpack a long, but then the pointer is stepped forward as if it were a double.  Looks like maybe this was supposed to be SIZEOF_NETWORK_LONG?  There's another one below too.

Reply:
Yeah the `txt_size` and `txt_rot_angle` are of double types so it should be deserialized/serialized using `bu_cv_ntohd()` / `bu_cv_htond()` instead of `ntohl()` / `htonl`.